src/networking.c

```diff
@@ -825,17 +825,18 @@ void addReplyLongLongWithPrefix(client *c, long long ll, char prefix) {
      * so we have a few shared objects to use if the integer is small
      * like it is most of the times. */
     const int opt_hdr = ll < OBJ_SHARED_BULKHDR_LEN && ll >= 0;
+    const size_t hdr_len = OBJ_SHARED_HDR_STRLEN(ll);
     if (prefix == '*' && opt_hdr) {
-        addReply(c,shared.mbulkhdr[ll]);
+        addReplyProto(c,shared.mbulkhdr[ll]->ptr,hdr_len);
         return;
     } else if (prefix == '$' && opt_hdr) {
-        addReply(c,shared.bulkhdr[ll]);
+        addReplyProto(c,shared.bulkhdr[ll]->ptr,hdr_len);
         return;
     } else if (prefix == '%' && opt_hdr) {
-        addReply(c,shared.maphdr[ll]);
+        addReplyProto(c,shared.maphdr[ll]->ptr,hdr_len);
         return;
     } else if (prefix == '~' && opt_hdr) {
-        addReply(c,shared.sethdr[ll]);
+        addReplyProto(c,shared.sethdr[ll]->ptr,hdr_len);
```

Comment:
i'm not certain this micro optimization is needed.

Reply:
@oranagra confirmed that it has an impact: from 851K ops/sec to 870K ops/sec


```
d95ae93fa4f2a8428d9cd8c43daba48135f7b079

BEST RUN RESULTS
==================================================================================================
Type         Ops/sec    Avg. Latency     p50 Latency     p99 Latency   p99.9 Latency       KB/sec 
--------------------------------------------------------------------------------------------------
Zranges    851602.48         3.74593         3.90300         4.67100         4.99100     70689.66 
Totals     851602.48         3.74593         3.90300         4.67100         4.99100     70689.66 


WORST RUN RESULTS
==================================================================================================
Type         Ops/sec    Avg. Latency     p50 Latency     p99 Latency   p99.9 Latency       KB/sec 
--------------------------------------------------------------------------------------------------
Zranges    850888.60         3.74906         3.91900         4.70300         5.11900     70630.40 
Totals     850888.60         3.74906         3.91900         4.70300         5.11900     70630.40 


AGGREGATED AVERAGE RESULTS (3 runs)
==================================================================================================
Type         Ops/sec    Avg. Latency     p50 Latency     p99 Latency   p99.9 Latency       KB/sec 
--------------------------------------------------------------------------------------------------
Zranges    851309.84         3.74723         3.90300         4.67100         5.05500     70665.37 
Totals     851309.84         3.74723         3.90300         4.67100         5.05500     70665.37 



6c0dada88308f20b084eeee01827dbe8064c8ae8


BEST RUN RESULTS
==================================================================================================
Type         Ops/sec    Avg. Latency     p50 Latency     p99 Latency   p99.9 Latency       KB/sec 
--------------------------------------------------------------------------------------------------
Zranges    878571.01         3.63070         3.83900         4.63900         5.02300     72928.26 
Totals     878571.01         3.63070         3.83900         4.63900         5.02300     72928.26 


WORST RUN RESULTS
==================================================================================================
Type         Ops/sec    Avg. Latency     p50 Latency     p99 Latency   p99.9 Latency       KB/sec 
--------------------------------------------------------------------------------------------------
Zranges    862275.15         3.69943         3.85500         4.63900         4.95900     71575.57 
Totals     862275.15         3.69943         3.85500         4.63900         4.95900     71575.57 


AGGREGATED AVERAGE RESULTS (3 runs)
==================================================================================================
Type         Ops/sec    Avg. Latency     p50 Latency     p99 Latency   p99.9 Latency       KB/sec 
--------------------------------------------------------------------------------------------------
Zranges    868237.75         3.67398         3.85500         4.63900         4.99100     72070.52 
Totals     868237.75         3.67398         3.85500         4.63900         4.99100     72070.52 
```